### ISSUE TITLE
nikola auto doesn't rebuild after exception

### ISSUE THREAD

**Issue by matclab** [getnikola/nikola#2940]
### Environment (if reporting a bug)
Python 3.6.4
**Python Version:** 3.6.4

**Nikola Version:** Nikola==7.8.11

**Operating System:** arch linux

### Description:

Running `nikola auto` mainly works, but after some time I got an exception:
```
[2017-12-31T10:34:18Z] INFO: auto: REFRESHING: output/categories/fr.xml
[2017-12-31T10:34:18Z] INFO: ws4py: Terminating websocket [Local => 127.0.0.1:8000 | Remote => 127.0.0.1:40164]
[2017-12-31T10:34:18Z] INFO: auto: ---> /categories/fr.xml
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib64/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
…
  File "/home/clabaut/.virtualenvs/nikola/lib/python3.6/site-packages/ws4py/websocket.py", line 283, in _write
    raise RuntimeError("Cannot send on a terminated websocket")
RuntimeError: Cannot send on a terminated websocket

[2017-12-31T10:34:18Z] INFO: ws4py: Terminating websocket [Local => 127.0.0.1:8000 | Remote => 127.0.0.1:40032]
[2017-12-31T10:34:18Z] INFO: ws4py: Managing websocket [Local => 127.0.0.1:8000 | Remote => 127.0.0.1:40704]
[2017-12-31T10:34:18Z] INFO: auto: <--- {'command': 'hello', 'protocols': ['http://livereload.com/protocols/official-6', 'http://livereload.com/protocols/official-7'], 'ver': '2.2.1', 'snipver': 1}
…
```
And then nikola does not refresh anymore. I have to stop and relaunch `nikola auto`.
It is not very important, but I've also observed this behavior in previous versions.

**Comment by Kwpolska** [comment]
`nikola auto` was rewritten from scratch for version 8. Can you try installing from GitHub master and see if it works better for you?

**Comment by matclab** [comment]
I've got a similar problem, but with a different exception:
```
[2017-12-31T11:27:13Z] INFO: auto: REFRESHING: output/index.html
[2017-12-31T11:27:13Z] INFO: auto: ---> /index.html
[2017-12-31T11:27:13Z] INFO: ws4py: Terminating websocket [Local => 127.0.0.1:8000 | Remote => 127.0.0.1:38226]
Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.6/site-packages/watchdog/observers/api.py", line 199, in run
    self.dispatch_events(self.event_queue, self.timeout)
  File "/usr/lib/python3.6/site-packages/watchdog/observers/api.py", line 368, in dispatch_events
    handler.dispatch(event)
  File "/usr/lib/python3.6/site-packages/watchdog/events.py", line 322, in dispatch
    self.on_any_event(event)
  File "/usr/lib/python3.6/site-packages/nikola/plugins/command/auto/__init__.py", line 433, in on_any_event
    self.function(event)
  File "/usr/lib/python3.6/site-packages/nikola/plugins/command/auto/__init__.py", line 293, in do_refresh
    refresh_signal.send(path=p)
  File "/usr/lib/python3.6/site-packages/blinker/base.py", line 267, in send
    for receiver in self.receivers_for(sender)]
  File "/usr/lib/python3.6/site-packages/blinker/base.py", line 267, in <listcomp>
    for receiver in self.receivers_for(sender)]
  File "/usr/lib/python3.6/site-packages/nikola/plugins/command/auto/__init__.py", line 405, in notify
    self.send(response, response.is_binary)
  File "/usr/lib/python3.6/site-packages/ws4py/websocket.py", line 307, in send
    self._write(data)
  File "/usr/lib/python3.6/site-packages/ws4py/websocket.py", line 285, in _write
    self.sock.sendall(b)
OSError: [Errno 9] Mauvais descripteur de fichier
```

Where `Mauvais descripteur de fichier` stands for `Bad file descriptor`.

**Comment by Kwpolska** [comment]
This is still Nikola v7, please try upgrading first.

**Comment by matclab** [comment]
Oups, you're right, I was in the wrong virtualenv, sorry.

I now get the following exception (still with python 3.6.4):

```
[2018-01-01T10:22:34Z] ERROR: asyncio: Task exception was never retrieved
future: <Task finished coro=<NikolaEventHandler.on_any_event() done, defined at /home/clabaut/.virtualenvs/nikola-dev/lib/python3.6/site-packages/nikola/plugins/command/auto/__init__.py:479> exception=AttributeError("'os.stat_result' object has no attribute 'st_file_attributes'",)>
Traceback (most recent call last):
  File "/home/clabaut/.virtualenvs/nikola-dev/lib/python3.6/site-packages/nikola/plugins/command/auto/__init__.py", line 482, in on_any_event
    yield from self.function(event)
  File "/home/clabaut/.virtualenvs/nikola-dev/lib/python3.6/site-packages/nikola/plugins/command/auto/__init__.py", line 277, in run_nikola_build
    is_hidden = os.stat(event_path).st_file_attributes & stat.FILE_ATTRIBUTE_HIDDEN
AttributeError: 'os.stat_result' object has no attribute 'st_file_attributes'
```

**Comment by Kwpolska** [comment]
`st_file_attributes` is Windows-specific. I fixed this on master (commit c7a556b5).

**Comment by matclab** [comment]
Thanks, I'll give a try for my next post.
